File: Code/C_Code/06.2.1_Alertor/Alertor.c
/**********************************************************************
* Filename    : Alertor.c
* Description : Alarm by button.
* Author      : freenove
* modification: 2016/06/14
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <softTone.h>
#include <math.h>

#define buzzerPin    0  	//define the buzzerPin
#define buttonPin 	 1		//define the buttonPin

void alertor(int pin){
	int x;
	double sinVal, toneVal;
	for(x=0;x<360;x++){ //frequency of the alarm along the sine wave change
		sinVal = sin(x * (M_PI / 180));		//calculate the sine value
		toneVal = 2000 + sinVal * 500;		//Add to the resonant frequency with a Weighted
		softToneWrite(pin,toneVal);			//output PWM
		delay(1);
	}
}
void stopAlertor(int pin){
	softToneWrite(pin,0);
}
int main(void)
{
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}	
	pinMode(buzzerPin, OUTPUT); 
	pinMode(buttonPin, INPUT);
	softToneCreate(buzzerPin);
	pullUpDnControl(buttonPin, PUD_UP);  //pull up to high level
	while(1){	
		if(digitalRead(buttonPin) == LOW){ //button has pressed down
			alertor(buzzerPin);   //buzzer on
			printf("alertor on...\n");
		}
		else {				//button has released 
			stopAlertor(buzzerPin);   //buzzer off
			printf("...buzzer off\n");
		}
	}
	return 0;
}

File: Code/Python_Code/06.2.1_Alertor/Alertor.py
#!/usr/bin/env python3
########################################################################
# Filename    : Alertor.py
# Description : Make Alertor with buzzer and button
# Author      : www.freenove.com
# modification: 2019/12/27
########################################################################
import RPi.GPIO as GPIO
import time
import math

buzzerPin = 11    # define the buzzerPin
buttonPin = 12    # define the buttonPin

def setup():
    global p    
    GPIO.setmode(GPIO.BOARD)         # Use PHYSICAL GPIO Numbering
    GPIO.setup(buzzerPin, GPIO.OUT)   # set RGBLED pins to OUTPUT mode
    GPIO.setup(buttonPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)    # Set buttonPin to INPUT mode, and pull up to HIGH level, 3.3V
    p = GPIO.PWM(buzzerPin, 1) 
    p.start(0);
    
def loop():
    while True:
        if GPIO.input(buttonPin)==GPIO.LOW:
            alertor()
            print ('alertor turned on >>> ')
        else :
            stopAlertor()
            print ('alertor turned off <<<')
def alertor():
    p.start(50)
    for x in range(0,361):      # Make frequency of the alertor consistent with the sine wave 
        sinVal = math.sin(x * (math.pi / 180.0))        # calculate the sine value
        toneVal = 2000 + sinVal * 500   # Add to the resonant frequency with a Weighted
        p.ChangeFrequency(toneVal)      # Change Frequency of PWM to toneVal
        time.sleep(0.001)
        
def stopAlertor():
    p.stop()
            
def destroy():
    GPIO.output(buzzerPin, GPIO.LOW)     # Turn off buzzer
    GPIO.cleanup()                       # Release GPIO resource

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()

